picard/ui/metadatabox/__init__.py

```diff
@@ -265,41 +295,134 @@ def keyPressEvent(self, event):
         else:
             super().keyPressEvent(event)
 
+    def get_selected_tags(self, items):
+        result = TagDiff()
+        for item in items:
+            tag, value = self._get_row_info(item.row())
+            col = item.column()
+            result.add(
+                tag=tag,
+                old=value[self.COLUMN_ORIG] if col == self.COLUMN_ORIG else None,
+                new=value[self.COLUMN_NEW] if col == self.COLUMN_NEW else None,
+                removable=self.tag_diff.status[tag] != TagStatus.NOTREMOVABLE,
+                readonly=self.tag_diff.status[tag] == TagStatus.READONLY
+            )
+
+        result.update_tag_names()
+        return result
+
+    def _get_row_info(self, row):
+        tag = self.tag_diff.tag_names[row]
+        value = {
+            self.COLUMN_ORIG: self.tag_diff.old[tag],
+            self.COLUMN_NEW: self.tag_diff.new[tag],
+        }
+        return tag, value
+
+    def _can_copy(self):
+        return True
+
     def _copy_value(self):
+        if not self.can_copy():
+            msg = N_("Unable to copy current selection.")
+            self.tagger.window.set_statusbar_message(msg, echo=log.info, timeout=3000)
+            return
+
+        items = self.selectedItems()
+        if len(items) > 1:
+            selected_data = self.get_selected_tags(items)
+            # Build the mimedata to use for the clipboard
+            mimedata = QtCore.QMimeData()
+            converted_data_cache = {}
+            for mimetype, converters in self.mimedatas.items():
+                if converters.encode_func:
+                    try:
+                        encoded_data = converted_data_cache.setdefault(mimetype, converters.encode_func(selected_data))
```

Comment:
Using `setdefault()` forces evaluation of `converters.encode_func(selected_data)`, do we really need a "cache" mechanism here?

Reply:
Data is converted to TSV for bother the TSV mimetype and the text mimetype. This caches is supposed to avoid that happening twice if the same encoding function is used for multiple mime-types.